### PR TITLE
core: frontend: store: autopilot: Add getters for parameters

### PR DESCRIPTION
**Pull request by patrickelectric**
A replace for #1395

As discussed on the dentist

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

**Comment by Williangalvani** [comment]
I think you need to declare things slightly differently in order for these to work as vuex getters.
check https://championswimmer.in/vuex-module-decorators/pages/core/getters.html and https://github.com/bluerobotics/BlueOS-docker/pull/1347/commits/cd4ba0435befd646edcdb8b9cae9106dd0cea263

It returns a lambda that takes the parameters

**Comment by patrickelectric** [comment]
@Williangalvani check now

File: core/frontend/src/store/autopilot.ts [hunk 1]
```diff
@@ -28,6 +28,18 @@ class AutopilotStore extends VuexModule {
 
   metadata_loaded = false
 
+  get parameter() {
+    // eslint-disable-next-line max-len
+    return (name: string): Parameter | undefined => this.parameters.filter((parameter) => parameter.name === name).first()
```
Comment (Williangalvani) [inline core/frontend/src/store/autopilot.ts:33]:
```suggestion
    return (name: string): Parameter | undefined => this.parameters.find((parameter) => parameter.name === name)
```

**Review by Williangalvani — CHANGES_REQUESTED** [review]
I think we should have a "filter" option that returns a list.  This is useful in cases such as "Get me all the SERVOXX_FUNCTION" parameters"

**Comment by patrickelectric** [comment]
> I think we should have a "filter" option that returns a list. This is useful in cases such as "Get me all the SERVOXX_FUNCTION" parameters". I'm not sure if it should be the regex one in this pr or new one that returns multiple elements

The regex on returns multiple objects

File: core/frontend/src/store/autopilot.ts (hunk 1 above)
Comment (patrickelectric) [reply core/frontend/src/store/autopilot.ts]: Nice

**Comment by patrickelectric** [comment]
> > I think we should have a "filter" option that returns a list. This is useful in cases such as "Get me all the SERVOXX_FUNCTION" parameters". I'm not sure if it should be the regex one in this pr or new one that returns multiple elements
> 
> The regex on returns multiple objects

Ops, my fault, fixing it

**Review by Williangalvani — APPROVED** [review]

**Pull request merged**